tensorflow/compiler/mlir/README.md

```diff
@@ -19,27 +19,17 @@ Building dialects and utilities here follow the standard approach using
 
 To develop across MLIR core and TensorFlow, it is useful to override the repo
 to use a local version instead of fetching from head. This can be achieved by
-setting up your local repository for Bazel build. For this you will need a
-temporary directory that will be "overlaid" with you LLVM source directory and
-the Bazel files:
+setting up your local repository for Bazel build. For this you will need to
+create bazel workspace and build files:
 
 ```sh
 LLVM_SRC=... # this the path to the LLVM local source directory you intend to use.
-LLVM_BAZEL_OVERLAY=${LLVM_SRC}/bazel # Note: this can be anywhere
-mkdir -p ${LLVM_BAZEL_OVERLAY}
-# This will symlink your LLVM sources with the BUILD files to be usable by Bazel.
-python ${LLVM_SRC}/utils/bazel/overlay_directories.py \
-    --src ${LLVM_SRC} \
-    --overlay ${LLVM_SRC}/utils/bazel/llvm-project-overlay/ \
-    --target ${LLVM_BAZEL_OVERLAY}
-touch ${LLVM_BAZEL_OVERLAY}/BUILD.bazel ${LLVM_BAZEL_OVERLAY}/WORKSPACE
-# The complete list is "AArch64", "AMDGPU", "ARM", "NVPTX", "PowerPC", "RISCV", "SystemZ", "X86"
-echo 'llvm_targets = ["X86"]' > ${LLVM_BAZEL_OVERLAY}/llvm/targets.bzl
+touch ${LLVM_SRC}/BUILD.bazel ${LLVM_SRC}/WORKSPACE
```

Comment:
Have you confirmed this is indeed necessary? I've never understood why external repositories need a WORKSPACE file that then goes entirely unused...

Reply:
Yeah there's just no reason they really need to be for dependencies. Everything would function just fine if there wasn't one there except for some reason Bazel decides to throw an error.

> As external repositories are repositories themselves, they often contain a WORKSPACE file as well. However, these additional WORKSPACE files are ignored by Bazel. In particular, repositories depended upon transitively are not added automatically.

Basically, I think [this check](https://github.com/bazelbuild/bazel/blob/844b5d2dbc3ff1dda43d78b81a00c6323e4e8115/src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java#L193) could be deleted with the only consequence being that people wouldn't have to create useless empty WORKSPACE files. Anyway, this is a total tangent about Bazel's design.